src/der.rs

```diff
@@ -212,6 +214,46 @@ pub(crate) fn read_tag_and_get_value_limited<'a>(
     Ok((tag, inner))
 }
 
+/// Prepend `bytes` with the given ASN.1 [`Tag`] and appropriately encoded length byte(s).
+/// Useful for "adding back" ASN.1 bytes to parsed content.
+#[cfg(feature = "alloc")]
+pub(crate) fn asn1_wrap(tag: Tag, bytes: &[u8]) -> Vec<u8> {
```

Comment:
This crate is more strict about `as` conversions than Rustls is. The ~3 uses in this function seem OK to me, so I would be OK with seeing a `allow(clippy::as_conversions)` annotation added to the fn to silence the warnings.

Alternatively this fn could be made fallible and the `as` conversions converted to use `try_into()`.

Reply:
Gotcha. I updated one of the `as` conversions (`u8` to `usize`) to a typical `into()`.
I think the other two `as` conversions are permissible given the conditional + assertion we have, so I added a `#[allow(clippy::as_conversions)]` annotation for those.

If anyone feels strongly about this, I can remove the annotation and replace the conversions with fallible ones, which can also act as a kind of runtime assertion.